tests/template_tests/test_base.py

```diff
@@ -84,3 +84,25 @@ def test_nonliterals(self):
         for var in var_names:
             with self.subTest(var=var):
                 self.assertIsNone(Variable(var).literal)
+
+    def test_scientific_notation_in_template(self):
```

Comment:
I don't think this test is needed given `test_filter_numeric_argument_parsing`
Also, this test case is around testing `Variable`

Reply:
agreed, this was from the original PR :)